tests/foreman/cli/test_ansible.py

```diff
@@ -677,3 +679,273 @@ def _finalize():
         )
         result = target_sat.cli.JobInvocation.info({'id': job_id})['success']
         assert result == '1'
+
+
+@pytest.mark.upgrade
+class TestAnsibleAAPIntegration:
+    """Test class for Satellite integration with Ansible Automation Controller
+
+    :CaseComponent: Ansible-ConfigurationManagement
+    """
+
+    @pytest.fixture(scope='class')
+    def aap_client(self):
+        client = awxkit.api.client.Connection(f'https://{settings.AAP_INTEGRATION.AAP_FQDN}/')
+        client.login(settings.AAP_INTEGRATION.USERNAME, settings.AAP_INTEGRATION.PASSWORD)
+        yield client
+        client.logout()
+
+    @pytest.mark.parametrize('auth_type', ['admin', 'non-admin'])
+    @pytest.mark.rhel_ver_list([settings.content_host.default_rhel_version])
+    def test_positive_ansible_dynamic_inventory(
+        self,
+        request,
+        target_sat,
+        module_org,
+        module_location,
+        module_ak_with_cv,
+        aap_client,
+        rhel_contenthost,
+        auth_type,
+    ):
+        """Verify AAP is able to pull a dynamic inventory of hosts from Satellite,
+        with admin and non-admin user.
+
+        :id: ad25aee5-4ea3-4743-a301-1c6271856g19
+
+        :steps:
+            1. Register content host to Satellite with admin and non-admin user
+            2. Update Satellite type credentials in AAP
+            3. Find Satellite inventory, inventory source and sync inventory source in AAP
+            4. Verify registered hosts are added to Satellite inventory
+
+        :expectedresults: All hosts managed by Satellite are added to Satellite inventory.
+        """
+        creds_name = settings.AAP_INTEGRATION.satellite_credentials
+        inventory_name = settings.AAP_INTEGRATION.satellite_inventory
+
+        password = settings.server.admin_password
+        if auth_type == 'admin':
+            login = gen_string('alpha')
```

Comment:
this feels like something that belongs to the `setup` phase rather to the body of the test. Could we possible move this to a fixture that would accept `role` as a parameter and request it by test by passing it the value of `auth_type` [indirectly](https://docs.pytest.org/en/stable/example/parametrize.html#indirect-parametrization)?

Reply:
@rplevka yeah, this is a setup phase, but this is only used by this test at this moment, so I would rather skip moving this to fixture.
This is a good point, I had considered such a common fixture in the past, but I don't recall the exact details, but I had some problems implementing it. So I will take a look at it again and get back to this as we have a couple of more tests that use this setup, but those will be done in separate PR.

Reply:
Ok. so i'll at least add some nitpick comments to the current implementation:
```
login = gen_string('alpha')
```
is redundant - you keep it in bot branches. It can be set once, outside of the conditional and used by both branches later.